maisim/maisim.Game/Graphics/UserInterfaceV2/BeatmapSetSelection.cs

```diff
@@ -1,59 +1,173 @@
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osuTK;
+
+namespace maisim.Game.Graphics.UserInterfaceV2
+{
+    public partial class BeatmapSetSelection : CompositeDrawable
+    {
+        [Resolved]
+        private CurrentWorkingBeatmap currentWorkingBeatmap { get; set; }
+
+        [Resolved]
+        private MusicPlayer musicPlayer { get; set; }
+
+        private List<Drawable> beatmapSetDrawables;
+        private Box topBox;
+        private Box bottomBox;
+        private Container dummyBox;
+        private Box backgroundBox;
+        private BasicScrollContainer scrollContainer;
+
+        private void workingBeatmapChange(ValueChangedEvent<BeatmapSet> beatmapSetEvent)
+        {
+            if (scrollContainer.IsDragged || scrollContainer.IsHovered)
+                musicPlayer.SeekTo(beatmapSetEvent.NewValue.PreviewTime);
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            beatmapSetDrawables = new List<Drawable>();
+
+            // Add BeatmapSetCard to the container
+            foreach (TestUtil.AvailableTrackMetadata availableTrackMetadata in TestUtil.AVAILABLE_BEATMAP_SET_TRACK)
+            {
+                if (availableTrackMetadata != TestUtil.AvailableTrackMetadata.None)
+                {
+                    BeatmapSet beatmapSet = new BeatmapSetTestFixture(availableTrackMetadata).BeatmapSet;
+                    beatmapSetDrawables.Add(new BeatmapSetCard(beatmapSet)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(0.6f, 0.6f),
+                        // Add a margin between each BeatmapSetCard
+                        Margin = new MarginPadding()
+                        {
+                            Top = 10
+                        }
+                    });
+                }
+            }
+
+            // This container need to be wide equal to the screen width to prevent the scroll bar from being cut off.
+            Anchor = Anchor.TopLeft;
+            Origin = Anchor.TopLeft;
+            RelativeSizeAxes = Axes.Y;
+            Position = new Vector2(120, 0);
+            Size = new Vector2(400, 1);
+            InternalChildren = new Drawable[]
+            {
+                backgroundBox = new Box
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = MaisimColour.SongSelectionContainerColor,
+                    Alpha = 0.5f
+                },
+                new Container
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        dummyBox = new Container()

```

Comment:
not sure dummyBox is right? am open to bikeshedding here, anything but "dummy" please